objective-c/route_guide/README.md

```diff
@@ -0,0 +1,280 @@
+
+...
+
+RTGRouteGuide *client = [[RTGRouteGuide alloc] initWithHost:kHostAddress];
+```
+
+Notice that we've specified the HTTP scheme in the host address. This is because the server we will be using to test our client doesn't use [TLS](http://en.wikipedia.org/wiki/Transport_Layer_Security). This is fine because it will be running locally on our development machine. The most common case, though, is connecting with a gRPC server on the internet, running gRPC over TLS. For that case, the HTTPS scheme can be specified (or no scheme at all, as HTTPS is the default value). The default value of the port is that of the scheme selected: 443 for HTTPS and 80 for HTTP.
+
+
+### Calling service methods
+
+Now let's look at how we call our service methods. As you will see, all these methods are asynchronous, so you can call them from the main thread of your app without worrying about freezing your UI or the OS killing your app.
+
+#### Simple RPC
+
+Calling the simple RPC `GetFeature` is nearly as straightforward as calling any other aynschronous method on Cocoa.
+
+```objective-c
+RTGPoint *point = [RTGPoint message];
+point.latitude = 40E7;
+point.longitude = -74E7;
+
+[client getFeatureWithRequest:point handler:^(RTGFeature *response, NSError *error) {
+  if (response) {
+    // Successful response received
+  } else {
+    // RPC error
+  }
+}];
+```
+
+As you can see, we create and populate a request protocol buffer object (in our case `RTGPoint`). Then, we call the method on the client object, passing it the request, and a block to handle the response (or any RPC error). If the RPC finished successfully, the handler block is called with a `nil` error argument, and we can read the response information from the server from the response argument. If, instead, some RPC error happened, the handler block is called with a `nil` response argument, and we can read the details of the problem from the error argument.
+
+```objective-c
+NSLog(@"Found feature called %@ at %@.", response.name, response.location);
+```
+
+#### Streaming RPCs
+
+Now let's look at our streaming methods. Here's where we call the response-streaming method `ListFeatures`, which results in our client receiving a stream of geographical `RTGFeature`s:
+
+```objective-c
+[client listFeaturesWithRequest:rectangle handler:^(BOOL done, RTGFeature *response, NSError *error) {
+  if (response) {
+    // Element of the stream of responses received
+  } else if (error) {
+    // RPC error; the stream is over.
+  }
+  if (done) {
+    // The stream is over (all the responses were received, or an error occured). Do any cleanup.
+  }
+}];
+```
+
+Notice how the signature of the handler block now includes a `BOOL done` parameter. The block will be called any number of times; only on the last call will the `done` argument be `YES`. If an error occurs, the RPC will be finished and the handler will be called with arguments `(YES, nil, error)`.
```

Comment:
The handler block can be called any number of times; only on the last call is the `done` argument value set to `YES`. If an error occurs, the RPC finishes and the handler is called with the arguments `(YES, nil, error)`.

(I know what you're getting at, but in general I like to keep things in a sort of continuous present. Otherwise you can end up in a situation of peculiarly wandering tenses, where one minute you're in the past, then in the future, then conditional, etc. etc.)